### DISCUSSION TITLE
xremap changes ISO_Level3_Shift to Alt_R even when remap section is empty

### DISCUSSION THREAD

**Discussion by aleloi** [xremap/xremap#605] [Q&A]
I want to use xremap for switching around modifier keys, and xkb layout variants for the base layout. I found that whenever xremap is running, ISO_Level3_Shift turns into Alt_R even when no remapping is defined. I've looked at related issues  #296 and #120, but think that my problem is different.

I tried running xremap with an empty config and expected it to do nothing:

```yml
modmap:
  - name: main remaps
    remap:
```

Running with RUST_LOG=debug printed this:
```log
[2023-10-06T21:26:06Z DEBUG xremap::event_handler] => 1: KEY_RIGHTALT
[2023-10-06T21:26:06Z DEBUG xremap::action_dispatcher] 1: KEY_RIGHTALT
[2023-10-06T21:26:06Z DEBUG xremap::event_handler] => 0: KEY_RIGHTALT
[2023-10-06T21:26:06Z DEBUG xremap::action_dispatcher] 0: KEY_RIGHTALT
```

My xkb layout variant contains the line `include "level3(ralt_switch)"`. It's supposed to turn my Alt_R into an extra modifier key. With it I can enter 4 symbols from every key instead of the usual 2. When I hit the right alt key, and xremap is not running, `xev` prints this:

```log
KeyPress event, serial 36, synthetic NO, window 0x2800001,
    root 0x50e, subw 0x0, time 27902179, (536,632), root:(2456,1738),
    state 0x0, keycode 108 (keysym 0xfe03, ISO_Level3_Shift ), same_screen YES,
    XKeysymToKeycode returns keycode: 92
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 36, synthetic NO, window 0x2800001,
    root 0x50e, subw 0x0, time 27902347, (536,632), root:(2456,1738),
    state 0x80, keycode 108 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XKeysymToKeycode returns keycode: 92
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```

When xremap starts, it looks like this:
```log

KeyPress event, serial 32, synthetic NO, window 0x2800001,
    root 0x50e, subw 0x0, time 27832053, (-1778,801), root:(142,1907),
    state 0x0, keycode 108 (keysym 0xffea, Alt_R), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 35, synthetic NO, window 0x2800001,
    root 0x50e, subw 0x0, time 27832148, (-1778,801), root:(142,1907),
    state 0x8, keycode 108 (keysym 0xffea, Alt_R), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```

I hope it's a fixable bug, and that xremap can forward key events exactly as they are when they are not configured to be remapped.

**Comment by k0kubun** [comment]
I suspect that it's because xremap grabs a device and creates its own device. I think you can apply the same xkb layout to xremap's device instead. For myself, I run `setxkbmap` _after_ starting xremap, and it works just fine. From what I'm hearing, you're probably changing the layout _before_ xremap starts, which might be the problem.
